### DISCUSSION TITLE
How to redirect time's output?

### DISCUSSION THREAD

**Discussion by Catizard** [fish-shell/fish-shell#10255] [Q&A] [answered]
Hi, I'm writing a speed test for my program and I use `time` to show how long does it take.But I found I cannot do the things follow:`time ./a.out > tmp.txt`.I checked manual and discussions found nothing.

**Comment by faho** [comment, marked as answer]
You can't redirect the output of fish's time keyword, no.

If you want to profile a script, you can use fish's `--profile` commandline option, like

```fish
fish --profile /tmp/my.prof /path/to/your/script
```

This will then give you a full list of what took how long.

Alternatively, if you want to *benchmark* it, you can use https://github.com/sharkdp/hyperfine/.

**Reply by Catizard** [reply]
Thanks for learning an option --profile.
I previously packed `time xxx` in a script file and use `fish -c` to redirect it lol.
